apps/studio/components/interfaces/Storage/CreateBucketModal.tsx

```diff
@@ -340,154 +357,165 @@ const CreateBucketModal = () => {
-                              />
-                            </FormControl_Shadcn_>
+                            <div className="grid grid-cols-12 gap-x-2">
+                              <div className="col-span-8">
+                                <FormControl_Shadcn_>
+                                  <Input_Shadcn_
+                                    id="formatted_size_limit"
+                                    aria-label="File size limit"
+                                    type="number"
+                                    min={0}
+                                    placeholder="0"
+                                    {...field}
+                                  />
+                                </FormControl_Shadcn_>
+                              </div>
+                              <div className="col-span-4">
+                                <Select_Shadcn_
+                                  value={selectedUnit}
+                                  onValueChange={setSelectedUnit}
+                                >
+                                  <SelectTrigger_Shadcn_
+                                    aria-label="File size limit unit"
+                                    size="small"
+                                  >
+                                    <SelectValue_Shadcn_ asChild>
+                                      <>{selectedUnit}</>
+                                    </SelectValue_Shadcn_>
+                                  </SelectTrigger_Shadcn_>
+                                  <SelectContent_Shadcn_>
+                                    {Object.values(StorageSizeUnits).map((unit: string) => (
+                                      <SelectItem_Shadcn_
+                                        key={unit}
+                                        value={unit}
+                                        className="text-xs"
+                                      >
+                                        <div>{unit}</div>
+                                      </SelectItem_Shadcn_>
+                                    ))}
+                                  </SelectContent_Shadcn_>
+                                </Select_Shadcn_>
+                              </div>
+                            </div>
                           </FormItemLayout>
                         )}
                       />
-                    </CollapsibleContent_Shadcn_>
-                  </Collapsible_Shadcn_>
+                      {formattedSizeLimitError?.message === 'exceed_global_limit' && (
+                        <FormMessage_Shadcn_ className="mt-2">
+                          Exceeds global limit of {formattedGlobalUploadLimit}. Increase limit in{' '}
+                          <InlineLink
+                            className="text-destructive decoration-destructive-500 hover:decoration-destructive"
+                            href={`/project/${ref}/settings/storage`}
```

Comment:
just realised - all these should be pointing to storage/settings instead (settings/storage is an old URL, so redirect to this URL via Link was causing a full re-render)

i'll get these fixed while i review btw no worries 🙂 🙏